include-build/roborazzi-core/src/commonJvmMain/kotlin/com/github/takahirom/roborazzi/CaptureResult.kt

```diff
@@ -1,124 +1,96 @@
-      json.put("timestamp", timestampNs)
-      return json
-    }
-  }
+  ) : CaptureResult
 
   data class Changed(
+    @SerializedName("compare_file_path")
     override val compareFile: File,
+    @SerializedName("golden_file_path")
     override val goldenFile: File,
+    @SerializedName("actual_file_path")
     override val actualFile: File,
+    @SerializedName("timestamp")
     override val timestampNs: Long
-  ) : CaptureResult {
-    override fun toJson(): JSONObject {
-      val json = JSONObject()
-      json.put("type", "changed")
-      json.put("compare_file_path", compareFile.absolutePath)
-      json.put("actual_file_path", actualFile.absolutePath)
-      json.put("golden_file_path", goldenFile.absolutePath)
-      json.put("timestamp", timestampNs)
-      return json
-    }
-  }
+  ) : CaptureResult
 
   data class Unchanged(
+    @SerializedName("golden_file_path")
     override val goldenFile: File,
+    @SerializedName("timestamp")
     override val timestampNs: Long
   ) : CaptureResult {
     override val actualFile: File?
       get() = null
     override val compareFile: File?
       get() = null
-
-    override fun toJson(): JSONObject {
-      val json = JSONObject()
-      json.put("type", "unchanged")
-      json.put("golden_file_path", goldenFile.absolutePath)
-      json.put("timestamp", timestampNs)
-      return json
-    }
   }
 
   companion object {
-    fun fromJsonFile(inputPath: String): CaptureResult {
-      val json = JSONObject(FileReader(inputPath).readText())
-      return fromJson(json)
+    fun fromJsonFile(filePath: String): CaptureResult {
+      return gson.fromJson(FileReader(filePath), CaptureResult::class.java)
     }
+  }
 
-    fun fromJson(json: JSONObject): CaptureResult {
-      val type = json.getString("type")
-      val compareFile = json.optString("compare_file_path")?.let { File(it) }
-      val goldenFile = json.optString("golden_file_path")?.let { File(it) }
-      val actualFile = json.optString("actual_file_path")?.let { File(it) }
-      val timestampNs = json.getLong("timestamp")
+  object JsonAdapter : com.google.gson.JsonSerializer<CaptureResult>,
+    com.google.gson.JsonDeserializer<CaptureResult> {
+    override fun serialize(
+      src: CaptureResult,
+      typeOfSrc: java.lang.reflect.Type,
+      context: com.google.gson.JsonSerializationContext
+    ): com.google.gson.JsonElement {
+      return when (src) {
+        is Recorded -> context.serialize(src, Recorded::class.java)
+        is Changed -> context.serialize(src, Changed::class.java)
+        is Unchanged -> context.serialize(src, Unchanged::class.java)
+        is Added -> context.serialize(src, Added::class.java)
+      }
+    }
 
+    override fun deserialize(
+      json: com.google.gson.JsonElement,
+      typeOfT: java.lang.reflect.Type,
+      context: com.google.gson.JsonDeserializationContext
+    ): CaptureResult {
+      val type = json.asJsonObject.get("type").asString
```

Comment:
```
...
Caused by: java.lang.NullPointerException: Cannot invoke "com.google.gson.JsonElement.getAsString()" because the return value of "com.google.gson.JsonObject.get(String)" is null
	at com.github.takahirom.roborazzi.CaptureResult$JsonAdapter.deserialize(CaptureResult.kt:88)
	at com.github.takahirom.roborazzi.CaptureResult$JsonAdapter.deserialize(CaptureResult.kt:68)
...
```
From this error message, it seems that error handling is missing when this part is null
```kotlin
val type = json.asJsonObject.get("type").asString
```